RLTest/__main__.py

```diff
@@ -298,38 +287,66 @@ def takeEnvDown(self, fullShutDown=False):
                     self.testsFailed.add(self.currEnv)
                 self.currEnv = None
 
-    def _runTest(self, method, printTestName=False, numberOfAssertionFailed=0):
+    def _runTest(self, test, printTestName=False, numberOfAssertionFailed=0):
+        if len(inspect.getargspec(test.target).args) > 0 and not test.is_method:
+            env = Env(testName=test.name)
+            fn = lambda: test.target(env)
+        else:
+            fn = test.target
+
         exceptionRaised = False
+        msgPrefix = test.name if not printTestName else ''
+
         if printTestName:
-            print '\t' + Colors.Cyan(method.__name__)
+            print '\t' + Colors.Cyan(test.name)
         try:
             if self.args.debug:
                 raw_input('\tenv is up, attach to any process with gdb and press any button to continue.')
```

Comment:
this is a but, the debug check should be on the env.py after setting up the env, if the test create the env by it own then he will get this message before the env is up, but this is an issue on master also so I will fix it.